### PR TITLE
fix(button): complete ripple when button becomes disabled

### LINKED ISSUE #4364: Retain existing ripples when button state changes to disabled
#### Bug, feature request, or proposal:
Bug

#### Expected behavior
When clicking on a button that will immediately become disabled, the ripple should remain until it fades out.

after

#### Current behavior
The ripple is immediately destroyed when the button state changes to disabled.

before

#### What are the steps to reproduce?
http://plnkr.co/edit/otcf7ejUS9GTY1s3s3v2?p=preview

#### Is there anything else we should know?
Happy to open a PR if this is desired behavior

### PR DESCRIPTION
**Pull request by willshowell**
Fixes #4364

### CODE REVIEW

File: src/lib/button/button.spec.ts [hunk 1]
```diff
@@ -156,41 +157,54 @@ describe('MdButton', () => {
...
-      buttonElement = fixture.nativeElement.querySelector('button[md-button]');
-      anchorElement = fixture.nativeElement.querySelector('a[md-button]');
+
+      buttonDebugElement = fixture.debugElement.query(By.css('button[md-button]'));
+      buttonRippleDebugElement = buttonDebugElement.query(By.directive(MdRipple));
+      buttonRippleInstance = buttonRippleDebugElement.injector.get<MdRipple>(MdRipple);
+
+      anchorDebugElement = fixture.debugElement.query(By.css('a[md-button]'));
+      anchorRippleDebugElement = anchorDebugElement.query(By.directive(MdRipple));
+      anchorRippleInstance = anchorRippleDebugElement.injector.get<MdRipple>(MdRipple);
```
Comment (DevVersion) [inline src/lib/button/button.spec.ts]: Do we really need to set the generic type here?

Comment (willshowell) [inline src/lib/button/button.spec.ts]: Getting directive instances was new to me, so I copied it from other material tests (radios, checkboxes, button-toggles, and sliders all use generic types for `injector.get()`)

Happy to remove if you want

Comment (DevVersion) [inline src/lib/button/button.spec.ts]: It's alright if other tests do it :)

**Comment by andrewseguin** [comment]
Looks like the tests aren't happy with you, can you see what's up?

**Comment by willshowell** [comment]
@andrewseguin they're passing again after rebase

**Comment by willshowell** [comment]
@jelbourn can `pr: merge ready` get added back now that tests are passing again?

**Comment by jelbourn** [comment]
One project has some failing e2e tests with this change, not quite sure why yet

**Pull request merged**